util/util_tls.go

```diff
@@ -0,0 +1,171 @@
+package util
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	apitypes "github.com/codedellemc/libstorage/api/types"
+)
+
+var (
+	orgName       = "libstorage"
+	certBlockType = "CERTIFICATE"
+	keyBlockType  = "RSA PRIVATE KEY"
+)
+
+func CreateSelfCert(ctx apitypes.Context, certPath, keyPath, host string) error {
+
+	// if files exist, ignore
+	_, cerErr := os.Stat(certPath)
+	_, keyErr := os.Stat(keyPath)
+	if cerErr == nil && keyErr == nil {
+		ctx.WithFields(log.Fields{
+			"host":     host,
+			"certPath": certPath,
+			"certKey":  certPath,
+		}).Debug("skipping self-cert creation, files exist")
+		return nil
+	}
+
+	certRoot := filepath.Dir(certPath)
+	keyRoot := filepath.Dir(keyPath)
+	if err := os.MkdirAll(certRoot, 0755); err != nil {
+		ctx.WithFields(log.Fields{
+			"host":     host,
+			"certRoot": certRoot,
+		}).Debug("created dir")
+
+		return err
+	}
+	if keyRoot != certRoot {
+		if err := os.MkdirAll(keyRoot, 0755); err != nil {
+			ctx.WithFields(log.Fields{
+				"host":    host,
+				"keyRoot": keyRoot,
+			}).Debug("created dir")
+			return err
+		}
+	}
+
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		return err
+	}
+
+	tmpl := x509.Certificate{
```

Comment:
Hi @vladimirvivien,

Please ensure all code is written with an 80 character boundary and wrapped accordingly with emphasis on readability. Thank you.